### ISSUE TITLE
Jupyter 4.0.1 - Unable to access filesystem outside of home directory/notebook directory

### ISSUE THREAD

**Issue by JoeSF49** [jupyterhub/jupyterhub#4493]
### Bug description
I've also posted to Discord.  please forgive me as this is my first bug report.  LMK the preferred method github v discord. ty

I’ve been using jupyterhub for a few years now. I’ve recently tried updating to 4.0.1 but it no longer seems to honor my Spawner.default_url and Spawner.notebook_dir. My system uses the base local single user spawner along with NativeAuthenticator. As of 4.0.1, I’m always locked within the home dir of the user which is not desirable as I have group dirs for users to share behind their home dirs. When I downgrade back to jupyterhub 3.1.1, I can move around as needed

I’ve also even tried this article [Troubleshooting — JupyterHub documentation](https://jupyterhub.readthedocs.io/en/stable/faq/troubleshooting.html#i-want-access-to-the-whole-filesystem-and-still-default-users-to-their-home-directory). This doesn’t appear to work at all as the redirects dup the users homedir path in the url

#### Expected behaviour

c.Spawner.notebook_dir should start at path specified.  Currently only drops user into homedir as "/" and therefore locked within that dir.  

#### Actual behaviour

c.Spawner.notebook_dir = ''/some/path"  should start the user at some specified path instead of forced homedir

### How to reproduce
user homedir ex....   /my_dir/username/nbs

install 4.x jupyter hub
set config:
c.Spawner.notebook_dir = "/my_dir"
c.Spawner.default_url = '/lab'

forces user into home '/my_dir/username'.  I want users to be able to go a level back for shared folders

reverting to Jupyterhub 3.1.1 fixes issue

### Your personal set up
Docker container cuda 11.8 ubuntu 20.08
running jupyterhub in conda base
default local spawner
NativeAuthenticator for auth

**Comment by JoeSF49** [comment]
I received an answer on discord thread.   c.Spawner.args=['--ContentsManager.root_dir=<dir_where_you_want_to_start_jlab>'].  fixed my issue

**Comment by JoeSF49** [comment]
on a side note where does this, ContentsManager, come from? A search on the documentation doesn’t yield anything… [Search - JupyterHub documentation](https://jupyterhub.readthedocs.io/en/stable/search.html?q=ContentsManager)

**Comment by mahendrapaipuri** [comment]
There is atleast one more [discussion](https://discourse.jupyter.org/t/c-spawner-notebook-dir-does-not-have-the-expected-effect/19913) regarding the same problem in JHub 4. I guess it is a side-effect of making singleuser-user app as a Jupyter server extension. 

I think it is worth documenting in the `notebook_dir` traitlet that users need to pass the same directory in `ContentsManager.root_dir` in `c.Spawner.args` for JHub 4.

**Comment by minrk** [comment]
This is indeed a bug in the singleuser extension ignoring config and fixed by #4503. One workaround is to opt-out of the singleuser extension with

```python
c.Spawner.environment = {
    "JUPYTERHUB_SINGLEUSER_EXTENSION": "0",
}
```

to stick with the mixin-based implementation.
